include/envoy/runtime/runtime.h

```diff
@@ -210,7 +210,8 @@ class Snapshot {
    * @param key supplies the key to fetch.
    * @return const std::string& the value or empty string if the key does not exist.
    */
-  virtual const std::string& get(absl::string_view key) const PURE;
+  virtual const std::string& get(absl::string_view key,
+                                 const std::string& default_value) const PURE;
```

Comment:
There is a potential issue with the return type of this function in the case where default_value is a temporary.  Take the following example:

const std::string& value = runtime.get("unknown_key", "default_value");
ENVOY_LOG_MISC(trace, "{}", value);

The default value string is deleted between the call to runtime.get and the call to ENVOY_LOG_MISC, leading to undefined behavior.